torchdynamo/variables/tensor.py

```diff
@@ -41,6 +44,16 @@
 from .lists import ShapeVariable
 from .lists import SizeVariable
 
+try:
+    import z3  # type: ignore[import]
```

Comment:
This code sounds familiar.  Perhaps refactor into:
`torchdynamo.utils.has_z3()`